flask_sqlalchemy/__init__.py

```diff
@@ -419,16 +419,50 @@ def first_or_404(self):
             abort(404)
         return rv
 
-    def paginate(self, page, per_page=20, error_out=True):
+    def paginate(self, page=None, per_page=None, error_out=True):
         """Returns `per_page` items from page `page`.  By default it will
         abort with 404 if no items were found and the page was larger than
         1.  This behavor can be disabled by setting `error_out` to `False`.
 
+        If page or per_page are None, they will be retrieved from the
+        request query.  If the values are not ints and ``error_out`` is
+        true, it will abort with 404.  If there is no request or they
+        aren't in the query, they default to page 1 and 20
+        respectively.
+
         Returns an :class:`Pagination` object.
         """
+
+        if has_request_context():
+            if page is None:
```

Comment:
@RonnyPfannschmidt you're absolutely right, I completely missed your comment. I'll create a new pr to address that.